logstash-core/src/main/java/org/logstash/ackedqueue/PageFactory.java

```diff
@@ -0,0 +1,80 @@
+package org.logstash.ackedqueue;
+
+import org.logstash.ackedqueue.io.PageIO;
+
+import java.util.BitSet;
+
+public class PageFactory {
+
+    /**
+     * create a new head page object and new page.{@literal {pageNum}} empty valid data file
+     *
+     * @param pageNum the new head page page number
+     * @param queue the {@link Queue} instance
+     * @param pageIO the {@link PageIO} delegate
+     * @return {@link Page} the new head page
+     */
+    public static Page newHeadPage(int pageNum, Queue queue, PageIO pageIO) {
+        return new Page(pageNum, queue, 0, 0, 0, new BitSet(), pageIO, true);
+    }
+
+    /**
+     * create a new head page from an existing {@link Checkpoint} and open page.{@literal {pageNum}} empty valid data file
+     *
+     * @param checkpoint existing head page {@link Checkpoint}
+     * @param queue the {@link Queue} instance
+     * @param pageIO the {@link PageIO} delegate
+     * @return {@link Page} the new head page
+     */
+    public static Page newHeadPage(Checkpoint checkpoint, Queue queue, PageIO pageIO) {
```

Comment:
@colinsurprenant the whole body of this method should probably be wrapped in a `try { ... } catch (Exception ex) {p.close; throw ex;}` Otherwise any exception thrown in the body will leave a `Page` behind that goes out of scope without a `close`.

Reply:
good idea.